app/scripts/metamask-controller.js

```diff
@@ -487,6 +494,12 @@ module.exports = class MetamaskController extends EventEmitter {
     await this.keyringController.submitPassword(password)
     const accounts = await this.keyringController.getAccounts()
 
+    // verify keyrings
+    const nonSimpleKeyrings = this.keyringController.keyrings.filter(keyring => keyring.type !== 'Simple Key Pair')
+    if (nonSimpleKeyrings.length > 1) {
+      if (this.diagnostics) await this.reportMultipleKeyrings(nonSimpleKeyrings)
```

Comment:
Nit, these compound `if` statements can be combined.

Reply:
fair